users/service.go

```diff
@@ -97,72 +120,72 @@ func (svc service) Register(ctx context.Context, session authn.Session, u User,
 	}()
 	user, err := svc.users.Save(ctx, u)
 	if err != nil {
-		return User{}, errors.Wrap(svcerr.ErrCreateEntity, err)
+		return User{}, svcerr.ErrFailedToSaveEntityDB.Embed(err)
```

Comment:
The original error orginate in repo layer , that was intentend to  expose in publish will be hidden by the service layers
Because service layer users embed, so the original public error which are orignated in repo layer will not  exposed


The  `err`  can have the following
- `errors.NewRequestError("email id already registered")`
- `errors.NewRequestError("username not available")`
These are intendented to expose to public , in API response